apis/apps/v1alpha1/cluster_types.go

```diff
@@ -293,6 +292,32 @@ type ReplicationMemberStatus struct {
 	Pod string `json:"pod"`
 }
 
+type CandidateInstance struct {
+	// index of the candidate instance, 0 <= index <= componentSpecs[x].replicas-1.
+	// +kubebuilder:validation:Minimum=0
+	// +kubebuilder:validation:Required
+	Index int32 `json:"index"`
```

Comment:
can check with XValidation rules for max range.

Reply:
> can check with XValidation rules for max range.

// ClusterComponentSpec defines the cluster component spec.
// +kubebuilder:validation:XValidation:rule="has(self.candidateInstance.index) && self.candidateInstance.index < self.replicas",message="candidateInstance.Index cannot be larger than replicas"

done